apps/mark-scan/backend/src/custom-paper-handler/state_machine.ts

```diff
@@ -686,9 +682,7 @@ export function buildMachine(
             },
 
             waiting_for_ballot_data: {
-              entry: assign({
-                isInVoterSession: true,
-              }),
+              invoke: pollPatDeviceConnectionStatus(),
```

Comment:
Just checking my understanding, we're now polling PAT device connection in specific voter states as opposed to all voter states, so specifically:
- `waiting_for_ballot_data`: The voting UI
- `presenting_ballot`: The confirmation UI
- `waiting_for_ballot_reinsertion`: That start of the reinsertion UI
- `waiting_for_invalidated_ballot_confirmation`: A follow-up step in the confirmation UI

Was the issue with the previous approach that we were polling PAT device connection status in too many states and surfacing the PAT configuration UI at points where it could interrupt operations like printing?

Reply:
That's correct. I'm not exactly sure why the transition guards didn't work in those cases, but we wanted to make this refactor anyway so I didn't spend too much time investigating.

Since the state machine has many states but only these 4 are voter states where the voter is in control, it made more sense to opt in to PAT polling than opt out with the guarded transition.